executor/executor.go

```diff
@@ -328,9 +330,43 @@ func (executor *Executor) execute(
 			}
 			return response, nil
 		}
-		msg := "decision executed successfully"
 
-		executor.logger.Infof(ctx, msg)
+		// short pooling to trigger pod status with max 15 minutes
+
+		statusMap := make(map[string]string)
+		statusMap["Running"] = "pod restarted successfully"
+		statusMap["Failed"] = "pod failed to restart"
+		statusMap["Unknown"] = "pod status is unknown"
+		statusMap["PodInitializing"] = "pod restarting"
+
+		backoff := 15 * time.Minute
+		msg := "pod restarting exceeded timout (15 min)"
+		start := time.Now()
+		timeout := int(backoff.Seconds())
+
+		for time.Now().Second() - start.Second() < timeout {
+
+			status := "Pending"
+
+			time.Sleep(15 * time.Second)
+			pods, _ := executor.kube.GetNameSpacePods(namespace)
+
+			for _, pod := range pods.Items {
+				if strings.Contains(pod.Name, name){
+					executor.logger.Info(pod.Name, ", status: ", pod.Status.Phase)
+					status = string(pod.Status.Phase)
+					break
+				}
+			}
+
+			if status != "Pending" {
+				msg = statusMap[status]
+				break
```

Comment:
I think it would be better relying on kubernetes client on pod status phases than turning the pod status to string and checking it, the same thing applies on the above status map.

https://sourcegraph.com/github.com/kubernetes/api/-/blob/core/v1/types.go#L2422:6&tab=references

@mohamedveron